pkg/watch/file_watcher.go

```diff
@@ -0,0 +1,137 @@
+package watch
```

Comment:
@feloy, there's a bunch of things being added/moved under the `pkg/watch` directory. I humbly request you to please add comments to the code. Especially functions, even if they are internal. It feels like magic right now, and could be really difficult for others to comprehend what's going on.

Reply:
I've added some more comments to the functions. But note that I added into the PR comment links to Unix and Windows terminal settings reference, that will help you understand what it is about.

Reply:
I'm referring to below functions:
* `getKeyWatcher`
* `getKey`
specifically, what are the go routines doing?

Reply:
Thanks. I just read them and it makes a huge difference for me personally. The explanations are very helpful!